modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/maven/LightweightClassLoaderModelBuilder.java

```diff
@@ -50,4 +69,84 @@ public LightweightClassLoaderModelBuilder(File artifactFolder,
         .map(bundleDescriptor -> mavenClient.resolveBundleDescriptor(bundleDescriptor))
         .collect(toList());
   }
+
+  @Override
+  protected Map<BundleDescriptor, Set<BundleDescriptor>> doProcessAdditionalPluginLibraries(Plugin packagingPlugin) {
+    File temporaryPomFolder = new File(MuleFoldersUtil.getExecutionFolder(), "mavenTemp");
+    if (!temporaryPomFolder.exists() && !temporaryPomFolder.mkdir()) {
+      throw new MuleRuntimeException(createStaticMessage("Could not create temporary folder under "
+          + temporaryPomFolder.getAbsolutePath()));
+    }
+    Map<BundleDescriptor, Set<BundleDescriptor>> deployableArtifactAdditionalLibrariesMap =
+        super.doProcessAdditionalPluginLibraries(packagingPlugin);
+    Map<BundleDescriptor, Set<BundleDescriptor>> effectivePluginsAdditionalLibrariesMap =
+        new HashMap<>(deployableArtifactAdditionalLibrariesMap);
+    nonProvidedDependencies.stream()
+        .filter(bundleDependency -> MULE_PLUGIN.equals(bundleDependency.getDescriptor().getClassifier().orElse(null)))
+        .forEach(bundleDependency -> {
+          Model effectiveModel;
+          try {
+            effectiveModel =
+                mavenClient.getEffectiveModel(toFile(bundleDependency.getBundleUri().toURL()), of(temporaryPomFolder));
+          } catch (MalformedURLException e) {
+            throw new MuleRuntimeException(e);
+          }
+          Optional<Plugin> artifactPackagerPlugin = findArtifactPackagerPlugin(effectiveModel);
+          artifactPackagerPlugin.ifPresent(plugin -> {
+            Map<BundleDescriptor, Set<BundleDescriptor>> pluginAdditionalLibrariesMap =
+                super.doProcessAdditionalPluginLibraries(artifactPackagerPlugin.get());
+            pluginAdditionalLibrariesMap.forEach((pluginDescriptor, additionalLibraries) -> {
+              Set<BundleDescriptor> effectivePluginAdditionalLibraries = new HashSet<>();
+              if (deployableArtifactAdditionalLibrariesMap.containsKey(pluginDescriptor)) {
+                effectivePluginAdditionalLibraries.addAll(additionalLibraries.stream()
+                    .filter(additionalLibrary -> {
+                      boolean additionalLibraryDefinedAtDeployableArtifact =
+                          existsInLibrariesMap(deployableArtifactAdditionalLibrariesMap, pluginDescriptor, additionalLibrary);
+                      if (!additionalLibraryDefinedAtDeployableArtifact) {
+                        Optional<BundleDescriptor> additionalLibraryDefinedByAnotherPlugin =
+                            findLibraryInAdditionalLibrariesMap(effectivePluginsAdditionalLibrariesMap, pluginDescriptor,
```

Comment:
Check my comment on mule-maven-plugin tests.
For what i can understand, an app that imports other reusable apps will behave as one. Is that correct? 
Consider App "A" and reusable app "R".
If R depends on a plugin X, then A will be able to "see" that plugin X as if it were declared in it's own pom, right? 
What happens if both A and R declare the same plugin but with different versions? 

I think that what happens in those scenarios will impact on whether or not this change is correct.

Reply:
What's going to happen is that the newest version of the plugin will be used. Same as if the app and another plugin have the same dependency with different versions.

Reply:
But i think that is not what happens. 
If an app and a plugin declare the same dependency with different versions, both will live in their own classLoader. However, if that dependency is a mule-plugin, then, it will happen as you said and the newest one will prevail. 
@gsfernandes care to comment on this?

Reply:
That's exactly what happens. If you declare an AD Z_v1 for Plugin X and another AD Z_v2 for another Plugin Y then there's no problem at all. Z_v1 will be added to Plugin X CL and Z_v2 will be added to Plugin Y.